packages/arb-token-bridge-ui/src/components/TransferPanel/Routes/Route.tsx

```diff
@@ -96,10 +98,290 @@ function getBadges(badgeTypes: BadgeType | BadgeType[]) {
+          ) : gasCost ? (
+            <div className="flex items-center gap-1" aria-label="Route gas">
+              {gasCost
+                .map(({ gasCost, gasToken }) =>
+                  formatAmount(BigNumber.from(gasCost), {
+                    decimals: gasToken.decimals,
+                    symbol: gasToken.symbol
+                  })
+                )
+                .join(' and ')}
+              {gasEth && (
+                <div
+                  className={twMerge(
+                    'tabular-nums opacity-80',
+                    compactMode ? 'text-xs' : 'text-sm'
+                  )}
+                >
+                  (
+                  {formatUSD(
+                    ethToUSD(
+                      Number(utils.formatEther(BigNumber.from(gasEth.gasCost)))
+                    )
+                  )}
+                  )
+                </div>
+              )}
+            </div>
+          ) : (
+            <div aria-label="Route gas">{'N/A'}</div>
+          )}
+        </span>
+      </div>
+    </Tooltip>
+
+    {bridgeFee && (
+      <Tooltip content={'The fee the bridge takes'}>
+        <div className="flex items-center gap-1">
+          <Image
+            src="/icons/bridge.svg"
+            width={compactMode ? 12 : 15}
+            height={compactMode ? 12 : 15}
+            alt="bridge fee"
+          />
+          <span>
+            {formatAmount(BigNumber.from(bridgeFee.fee), {
+              decimals: bridgeFee.token.decimals,
+              symbol: bridgeFee.token.symbol
+            })}
+          </span>
+          {showUSDValueForBridgeFee && (
+            <div
+              className={twMerge(
+                'tabular-nums opacity-80',
+                compactMode ? 'text-xs' : 'text-sm'
+              )}
+            >
+              (
+              {formatUSD(
+                ethToUSD(
+                  Number(utils.formatEther(BigNumber.from(bridgeFee.fee)))
+                )
+              )}
+              )
+            </div>
+          )}
+        </div>
+      </Tooltip>
+    )}
+  </>
+)
+
+// Route Badge Component
+type RouteBadgeProps = {
+  tag?: BadgeType | BadgeType[]
+  compactMode: boolean
+}
+
+const RouteBadge = ({ tag, compactMode }: RouteBadgeProps) => {
+  if (!tag || compactMode) return null
+
+  return (
+    <div className="invisible absolute right-2 top-2 flex gap-1 md:visible">
+      {getBadges(tag)}
+    </div>
+  )
+}
+
+// Route Summary Badge Component
+type RouteSummaryBadgeProps = {
+  bridge: string
+  bridgeIconURI: string
+  durationMs: number
+}
+
+const RouteSummaryBadge = ({
+  bridge,
+  bridgeIconURI,
+  durationMs
+}: RouteSummaryBadgeProps) => (
+  <div className="flex items-center gap-1 rounded-full bg-white/20 px-2 py-1">
+    <Tooltip content={bridge}>
+      <SafeImage
+        src={bridgeIconURI}
+        width={12}
+        height={12}
+        alt="bridge"
+        className="h-4 w-4 rounded-full"
+      />
```

Comment:
I would add a fallback here, so image doesn't jump when it's loaded
![image](https://github.com/user-attachments/assets/2ccba937-9482-4220-a4a3-cc19dca74379)